walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,119 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    /**
+     * Given the duration for a meeting, a set of attendees required to 
+     * attend the meeting, and a set of events each with their own respective 
+     * set of attendees, find and return the set of valid meeting times. 
+     * Valid meeting times are time ranges such that the meeting time is 
+     * greater than the given duration and all attendees are available 
+     * during that given time. 
+     * 
+     * To find valid meeting times, we sort all events by latest end time
+     * first and then iterate through each event, effectively traveling 
+     * backwards. For each event, we check if the event's attendees overlap 
+     * with the meeting's required attendees. If they do, then we cap off our 
+     * current time range, which spans [event end time, prevEndTime), and add 
+     * it to the valid meeting times if it is longer than the given duration. 
```

Comment:
It would be good to indicate that are the properties of the 'current time range': it's a time range where all the attendees are available.